### DISCUSSION TITLE
Run installation of apps for multiple rasberry PI"s

### DISCUSSION THREAD

**Discussion by Pershanthenm** [epoptes/epoptes#94] [Q&A]
I am busy playing around with Epoptes and it is a really cool tool to manage Pi's in a network. I highly recommend you guys to try it out.

I do have a question and cant seem to find a work around but has anyone been able to run an installation command across multiple devices on Epoptes? Essentially what i want to do is install Chrome for example on all selected devices(Using one command) rather than installing them one by one on each device. If you were able to do this on epoptes please share.

**Comment by alkisg** [comment]
From the Epoptes execute command dialog:
sudo apt install --yes chromium-browser

**Comment by Pershanthenm** [comment]
I tried that but didnt seem to work.

**Comment by alkisg** [comment]
Try sudo xterm
Does it open an xterm?
If not, you didn't configure epoptes correctly.

**Comment by Pershanthenm** [comment]
Hey Alkis,


I trust you are well. I am not at work at the moment to confirm. I will try
tomorrow again. Thanks so much for your immediate response. It's much
appreciated. :-)


Kind regards

**Comment by Pershanthenm** [comment]
Hey Alkis,
I trust you are well. I tried installing firefox to try this out and its
not installing. When i try reboot on the command box it reboots all Pi's.
It looks like the installed correctly but the installation of apps does
not. any thoughts?

Kind Regards
Pershanthen Moodley

**Comment by alkisg** [comment]
Hi, please answer my previous comment.

**Comment by Pershanthenm** [comment]
Hey Alkis,

Unfortunately not. Can you give me some advice?

Pershanthen Moodley

**Comment by alkisg** [comment]
What is the output of this command on a client?

ps aux | grep epoptes

**Comment by alkisg** [comment]
*and, do you actually have xterm installed?

**Comment by Pershanthenm** [comment]
Hey Alkis,

I have tried running Xterm and the both clients have xterm installed. I ran
the command and nothing seemed to happen.

**Comment by alkisg** [comment]
It sounds like Epoptes isn't properly installed, and only the session service runs, not the system service.

What is the output of these commands on a client?
ps aux | grep epoptes
/usr/sbin/epoptes-client --version

**Comment by Pershanthenm** [comment]
Hey Alkis,

I have not heard from you for awhile. Will really appreciate some feedback.

**Comment by alkisg** [comment]
Pershanthen, please visit this page:
https://github.com/Epoptes/epoptes/issues/94

My reply is 2 hours ago. How can you miss it?

**Comment by Pershanthenm** [comment]
Hey Alkis, 

I am so sorry, My reply did not go through. Please see link below,

https://i.imgur.com/QEwvrnJ.png

**Comment by alkisg** [comment]
The problem is that you have an old Epoptes version, 0.5.10.
Raspbian doesn't run ifupdown scripts, so the Epoptes system service doesn't start, and you can't run `sudo` commands and you can't control the clients before they log in etc.
Only the user service starts, when the clients log in.

In the latest Epoptes, 1.0, a systemd unit is used, so the system Epoptes service will properly run.

I.e. just update your Epoptes to 1.0.
https://packages.debian.org/buster/epoptes-client

**Comment by Pershanthenm** [comment]
Hey Alkis,

Thanks so much for your help. I am having trouble updating my epoptes
client using that document. can you please provide additional documentation
or any short commands?

**Comment by alkisg** [comment]
Epoptes offers new versions for Ubuntu via a ppa:
http://www.epoptes.org/documentation/ppa

We can't offer new versions for all distributions, it's up to your distribution to get the newer version.
Raspbian gets its packages from Debian, and you can see from the [link I pasted](https://packages.debian.org/buster/epoptes-client) that Debian already has Epoptes 1.0.

Now you need to contact your distribution on how to get that newer version.
